thrust/thrust/detail/config/namespace.h

```diff
@@ -162,18 +157,22 @@
 #    define THRUST_DETAIL_ABI_NS_END
 #  else // not defined(THRUST_DISABLE_ABI_NAMESPACE)
 #    if defined(_NVHPC_CUDA)
-#      define THRUST_DETAIL_ABI_NS_BEGIN inline namespace THRUST_DETAIL_ABI_NS_NAME(THRUST_VERSION, NV_TARGET_SM_INTEGER_LIST) {
+#      define THRUST_DETAIL_ABI_NS_BEGIN                                    \
+        inline namespace THRUST_DETAIL_SPLICE_WITH_UNDERSCORE(              \
+          THRUST, THRUST_VERSION, SM, NV_TARGET_SM_INTEGER_LIST, NVHPC, NS) \
```

Comment:
important: `THRUST`, and `NS` used to be part of the macro. Now they occupy two slots in `__VA_ARGS__` with `SM` and `NVHPC` added as well. The macro supports up to 20 arguments. This means, we have 20 - 4 = 16 slots left. Current nvcc compiles for up to 15 architectures with `-arch=all`, which leaves us with one more slot to use until this macro breaks. This means that introduction of new architectures could break our code. I think it makes sense to have some room left just in case, so I'd consider extending the macro to support more parameters, or finding a way not to provide some of the non-arch parameters (like omitting SM and NS).

Reply:
> This means, we have 20 - 4 = 16 slots left. Current nvcc compiles for up to 15 architectures

Perfection :D :D

Alright, I will add more splots!